numpy/core/src/umath/_umath_tests.c.src

```diff
@@ -671,6 +721,42 @@ err:
     return NULL;
 }
 
+static int
+add_INT32_negative_indexed(PyObject *module, PyObject *dict) {
+    if (import_experimental_dtype_api(__EXPERIMENTAL_DTYPE_VERSION) < 0) {
+        return -1;
+    }
+
+    PyObject * negative = PyUFunc_FromFuncAndData(NULL, NULL, NULL, 0, 1, 1,
+                                    PyUFunc_Zero, "indexed_negative", NULL, 0);
+    if (negative == NULL) {
+        return -1;
+    }
+    PyArray_DTypeMeta *dtype = PyArray_DTypeFromTypeNum(NPY_INT);
+    PyArray_DTypeMeta *dtypes[] = {dtype, dtype};
```

Comment:
```suggestion
    PyArray_DTypeMeta *dtype = PyArray_DTypeFromTypeNum(NPY_INT);
    PyArray_DTypeMeta *dtypes[] = {&PyArray_Int32DType, &PyArray_Int32DType};
```
The experimental API is actually more convenient here, plus otherwise we probably need a decref technically...

I am hoping using `int32` here will also fix that test failure (but not sure).

Reply:
fixed a little differently, thanks.

Reply:
Fair, might lose a reference, but only once during init, which really doesn't matter.  Thanks